### PR TITLE
[Enhancement] Ensure mutable string configs are thread-safe

### PR DESCRIPTION
**Pull request by sduzh**
## Why I'm doing:
BE's allows the definition of string configurations that can be dynamically modified at runtime. In order to avoid multi-threaded contention issues, these mutable string configurations are protected by a global mutex lock `mstring_conf_lock`, which needs to be acquired before reading or writing mutable string configurations to ensure thread safety.
There are two problems with this:
1. it's easy to forget to acquire `mstring_conf_lock` before reading or writing mutable string config, which did happened right now.
2. The global lock `mstring_conf_lock` has a potential multi-threaded scability issue.

## What I'm doing:
Define mutable string config as a `MutableString` type and provide thread-safe read and write interfaces. Instead of explicitly acquiring a global lock, users can access mutable string config as if it were a normal variable.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -1020,9 +1020,9 @@ CONF_Int64(max_length_for_bitmap_function, "1000000");
 
 // Configuration items for datacache
 CONF_Bool(datacache_enable, "false");
-CONF_mString(datacache_mem_size, "10%");
-CONF_mString(datacache_disk_size, "0");
-CONF_mString(datacache_disk_path, "${STARROCKS_HOME}/datacache/");
+CONF_String(datacache_mem_size, "10%");
+CONF_String(datacache_disk_size, "0");
+CONF_String(datacache_disk_path, "${STARROCKS_HOME}/datacache/");
 CONF_String(datacache_meta_path, "${STARROCKS_HOME}/datacache/");
```
Comment (imay) [inline be/src/common/config.h:1026]: If you change mString to String, all immutable string configure can be changed too.
But I don't think this is safe for all configurations.

Comment (sduzh) [reply be/src/common/config.h:1026]: I changed thse three configurations to immutable configurations becauses they are actually immutable: modifying their values at runtime has no real effect.

Comment (sduzh) [reply be/src/common/config.h:1026]: I've already confirmed this with @GavinMar

Comment (sduzh) [reply be/src/common/config.h:1026]: > ... But I don't think this is safe for all configurations.

Could you please give a more detailed description, I don't quite understand what you mean.

**Review by kevincai — APPROVED** [review]

File: be/test/storage/lake/vacuum_test.cpp [hunk 2]
```diff
@@ -1311,7 +1311,7 @@ TEST(LakeVacuumTest2, test_delete_files_retry2) {
     ASSERT_OK(f1->append("111"));
     ASSERT_OK(f1->close());
 
-    auto backup = config::lake_vacuum_retry_pattern;
+    auto backup = config::lake_vacuum_retry_pattern.value();
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/test/storage/lake/vacuum_test.cpp:1314]: You agree with you, but I haven't come up with any good ideas so far.

**Review by imay — APPROVED** [review]

**Pull request merged**